### DISCUSSION TITLE
Remove input from all code cells in a notebook

### DISCUSSION THREAD

**Discussion by alimanfoo** [executablebooks/meta#503] [Ideas] [answered]
Hi folks,

I'm just exploring using jupyter book for [writing a report (article) from a single notebook](https://jupyterbook.org/customize/toc.html#structure-of-an-article), and I am expecting a common scenario will be to want to remove inputs from all code cells, i.e., just show outputs.

This can be achieved by [adding the "remove-input" tag](https://jupyterbook.org/interactive/hiding.html#remove-cell-inputs) to all code cells, but that is likely to get tedious to do by hand.

I also found [this example script](https://jupyterbook.org/content/metadata.html#add-tags-using-python-code) showing how to add tags to cells programmatically, which I could definitely adapt to achieve what I want.

I'm just thinking it could be nice to add built-in capability to do this, e.g., a command line parameter like `jupyter-book build path/to/report.ipynb --remove-input`. 

If we adopt jupyter book as our main technology for writing reports I expect I'll be teaching a few people how to use it, and would be nice not to have to also pass around a custom script to add the remove-input tags.

**Comment by chrisjsewell** [comment, marked as answer]
Heya, I'm part way through a bit of a rewrite of the myst-nb module (that handles this), and here I am looking to implement "cascading" configuration, e.g. prioritizing: CLI < _config.yml < notebook level metadata < cell level metadata

**Reply by alimanfoo** [reply]
Sounds great, thanks so much.

**Comment by akhmerov** [comment]
This is also tracked in https://github.com/executablebooks/jupyter-book/issues/666

In the meantime, you can make a sphinx extension that adds the desired tag to all cells. I use this approach [here](https://github.com/topocm/topocm_content/blob/2165b7f95bc59221b57b30a7215c39004c6cd40e/sphinxext/extensions.py#L72-L88), combined with [this](https://github.com/topocm/topocm_content/blob/2165b7f95bc59221b57b30a7215c39004c6cd40e/_config.yml#L29-L30).

**Reply by alimanfoo** [reply]
Great, thank you, somehow managed to miss that issue when I searched earlier.
